tests/scadnano_tests.py

```diff
@@ -91,6 +91,177 @@ def test_strand__loopouts_with_labels_to_json(self) -> None:
         self.assertEqual(1, len(design_from_json.strands))
         self.assertEqual(expected_strand, design_from_json.strands[0])
 
+    def test_strand__extension_3p_from_to(self) -> None:
+        design: sc.Design = sc.Design(helices=[sc.Helix(max_offset=100)])
+        sb = design.draw_strand(0, 0)
+        sb.to(10)
+
+        sb.extension(5)
+
+        expected_strand: sc.Strand = sc.Strand([
+            sc.Domain(0, True, 0, 10),
+            sc.Extension(5),
+        ])
+        self.assertEqual(1, len(design.strands))
+        self.assertEqual(expected_strand, design.strands[0])
+
+    def test_strand__extension_5p(self) -> None:
+        design: sc.Design = sc.Design(helices=[sc.Helix(max_offset=100)])
+        sb = design.draw_strand(0, 0)
+
+        sb.extension(5)
+        sb.to(10)
+
+        expected_strand: sc.Strand = sc.Strand([
+            sc.Extension(5),
+            sc.Domain(0, True, 0, 10),
+        ])
+
+        self.assertEqual(1, len(design.strands))
+        self.assertEqual(expected_strand, design.strands[0])
+
+    def test_strand__update_to_after_extension_5p_ok(self) -> None:
+        design: sc.Design = sc.Design(helices=[sc.Helix(max_offset=100)])
+        sb = design.draw_strand(0, 0)
+
+        sb.extension(5)
+        sb.to(10)
+        sb.update_to(15)
+
+        expected_strand: sc.Strand = sc.Strand([
+            sc.Extension(5),
+            sc.Domain(0, True, 0, 15),
+        ])
+
+        self.assertEqual(1, len(design.strands))
+        self.assertEqual(expected_strand, design.strands[0])
+
+    def test_strand__lone_extension_should_not_add_strand(self) -> None:
+        design: sc.Design = sc.Design(helices=[sc.Helix(max_offset=100)])
+        sb = design.draw_strand(0, 0)
+
+        sb.extension(5)
+
+        self.assertEqual(0, len(design.strands))
+
+    def test_strand__to_after_non_5p_extension_should_raise_error(self) -> None:
+        design: sc.Design = sc.Design(helices=[sc.Helix(max_offset=100)])
+        sb = design.draw_strand(0, 0)
+        sb.to(10)
+        sb.extension(5)
+
+        with self.assertRaises(sc.IllegalDesignError):

```

Comment:
I may have missed this, but are you also testing creating an extension an error for calling `to`, then `extension`, then `to`? Also `move`, then `extension`, then `move`.

Reply:
I have to, extension, to test: 

https://github.com/UC-Davis-molecular-computing/scadnano-python-package/blob/78c716db99695396fd475af72832bc68fc3805eb/tests/scadnano_tests.py#L147-L154

I'll add move, extension, move test

Reply:
here's move, extension, move test
https://github.com/UC-Davis-molecular-computing/scadnano-python-package/blob/2cdf0234fbf65f2ed4e8638ec256ece7b2d3f63b/tests/scadnano_tests.py#L171-L178